### PR TITLE
Volume related api docs

### PR DESCRIPTION
**Pull request by jnummelin**
Added API docs for volume APIs and updated stack parts related to volumes

### CODE REVIEW

File: server/docs/source/index.html.md [hunk 1]
```diff
@@ -1267,6 +1293,78 @@ For example `"secret_name": "FOO_DOMAIN_COM"` will write following secrets to th
...
+    "driver_specific_option": "foobar",
+    "another_option": "xyz"
+  }
+}
+```
+
+Attribute | Description
+--------- | -----------
+name      | Name of the volume
+scope     | Scope for the volume
```
Comment (SpComb) [inline server/docs/source/index.html.md]: List what the currently supported scopes are?

File: server/docs/source/index.html.md [hunk 2]
```diff
@@ -1267,6 +1293,78 @@ For example `"secret_name": "FOO_DOMAIN_COM"` will write following secrets to th
...
+driver    | Volume driver to be used
```
Comment (SpComb) [inline server/docs/source/index.html.md]: This could reference the `node['volume_drivers'][*]['name']`

**Comment by SpComb** [comment]
@jakolehm How experimental do we want the volumes support to be...?

1. The `/v1/volumes` API is stable, but the stacks `volumes` is limited to just `external` volumes
2. The `/v1/volumes` API and `kontena volume ...` CLI are documented as experimental
3. We go all the way and require `/v1.experimental/volumes` and `kontena --experimental volume ...`

**Comment by jnummelin** [comment]
IMO option 2 should be enough. Option 3 requires quite some work on the API level to support that properly.

File: server/docs/source/index.html.md [hunk 3]
```diff
@@ -1267,6 +1293,80 @@ For example `"secret_name": "FOO_DOMAIN_COM"` will write following secrets to th
...
 `POST /v1/certificates/{grid_id}/certificate`
 
+# Volumes
+
+**Volumes support is at experimental state which means that there might be breaking changes between versions until the experimental status is removed.**
+
+## Volume
+
+```json
+{
```
Comment (jakolehm) [inline server/docs/source/index.html.md]: Id missing?

File: server/docs/source/index.html.md [hunk 4]
```diff
@@ -1267,6 +1293,80 @@ For example `"secret_name": "FOO_DOMAIN_COM"` will write following secrets to th
...
+Creates a volume to a grid
+
+### Endpoint
+
+`POST /v1/volumes/{grid_id}`
+
+## Delete a volume
+
+```http
+DELETE /v1/volumes/{grid_id}/{volume_name} HTTP/1.1
```
Comment (jakolehm) [inline server/docs/source/index.html.md]: `DELETE /v1/volumes/{volume_id}`

File: server/docs/source/index.html.md [hunk 5]
```diff
@@ -1267,6 +1293,80 @@ For example `"secret_name": "FOO_DOMAIN_COM"` will write following secrets to th
...
+Authorization: bearer 8dqAd30DRrzzhJzbcSCG0Lb35csy5w0oNeT+8eDh4q2/NTeK3CmwMHuH4axcaxya+aNfSy1XMsqHP/NsTNy6mg==
+Accept: application/json
+```
+### Endpoint
+
+`DELETE /v1/volumes/{grid_id}/{volume_name}`
```
Comment (jakolehm) [inline server/docs/source/index.html.md]: `DELETE /v1/volumes/{volume_id}`

**Comment by jakolehm** [comment]
@SpComb 2. should be ok.

**Comment by jnummelin** [comment]
Added also fields from #2099 to the docs

File: server/docs/source/index.html.md [hunk 6]
```diff
@@ -1267,6 +1293,92 @@ For example `"secret_name": "FOO_DOMAIN_COM"` will write following secrets to th
...
+
+```http
+GET /v1/volumes/{grid_id} HTTP/1.1
+Authorization: bearer 8dqAd30DRrzzhJzbcSCG0Lb35csy5w0oNeT+8eDh4q2/NTeK3CmwMHuH4axcaxya+aNfSy1XMsqHP/NsTNy6mg==
+Accept: application/json
+```
+
+### Endpoint
+
+`GET /v1/volumes/{grid_id}`
```
Comment (SpComb) [inline server/docs/source/index.html.md]: I find the `GET /v1/volumes/{grid_id}` vs `GET /v1/volumes/:grid/:volume` surprising, but this is consistent with the other API endpoint docs.

File: server/docs/source/index.html.md [hunk 7]
```diff
@@ -523,6 +535,20 @@ version | A version number for the stack
...
+### Volume attributes
+
+Attribute | Description
+---------- | -------
+name  | Name of the volume within the stack
+external | How the volume name is mapped to grid volume
+
+### Volume external attribute
+
+When `"external":true` Kontena references the grid level volume with the same name as defined in the stack yaml.
```
Comment (SpComb) [inline server/docs/source/index.html.md]: Nope.

File: server/docs/source/index.html.md [hunk 8]
```diff
@@ -523,6 +535,20 @@ version | A version number for the stack
...
+
+When `"external": {"name": "otherVolume"}` Kontena uses a grid volume called `otherVolume`
```
Comment (SpComb) [inline server/docs/source/index.html.md]: With `{"external": "otherVolume"}`.

File: server/docs/source/index.html.md [hunk 9]
```diff
@@ -508,6 +508,18 @@ to | The end date and time (example: `?to=2017-01-01T13:15:00.00Z`) | now
   			]
   		}
   	}
+  ],
+  "volumes": [
+    {
+      "name": "aVolume",
+      "external": true
```
Comment (SpComb) [inline server/docs/source/index.html.md]: Nope

File: server/docs/source/index.html.md [hunk 10]
```diff
@@ -508,6 +508,18 @@ to | The end date and time (example: `?to=2017-01-01T13:15:00.00Z`) | now
...
+    },
+    {
+      "name": "otherVolume",
+      "external": {
```
Comment (SpComb) [inline server/docs/source/index.html.md]: `"external": "otherName",`

File: server/docs/source/index.html.md (hunk 3 above)
Comment (jnummelin) [inline server/docs/source/index.html.md]: not anymore :)

**Pull request merged**